gluten-data/src/main/scala/io/glutenproject/metrics/FileSourceScanMetricsUpdater.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.metrics
+
+import org.apache.spark.sql.execution.metric.SQLMetric
+import org.apache.spark.sql.utils.OASPackageBridge.InputMetricsWrapper
+
+class FileSourceScanMetricsUpdater(
+    @transient val metrics: Map[String, SQLMetric]) extends MetricsUpdater {
```

Comment:
Would you like to add some notes here so that we can remember why @transient is used?

e.g.

```
// "val metrics" is made transient to avoid sending driver-side metrics to tasks, e.g. "pruning time" from scan.
```